File: golang/algorithms/DefangedIpAddress.go
package algorithms 

import (
	"fmt"
	"bytes"
	"strings"
)

// Description: https://leetcode.com/problems/defanging-an-ip-address/

type DefangedIpAddress struct {}

func (d *DefangedIpAddress) Defang(input string) string {
	var buffer bytes.Buffer
	for _, char := range input {
		if char == '.' {
			buffer.WriteString("[.]")
		} else {
			buffer.WriteString(string(char))
		}
	}
	return buffer.String()
}

func (d *DefangedIpAddress) Execute() {
	fmt.Println(strings.ToUpper("Defanging an IP-address\n"))

	// Example 1:
	// Input: address = "1.1.1.1"
	// Output: "1[.]1[.]1[.]1"
	var input1 string = "1.1.1.1"
	output1 := d.Defang(input1)
	fmt.Println("input:", input1, "output:", output1)

	// Example 2:
	// Input: address = "255.100.50.0"
	// Output: "255[.]100[.]50[.]0"
	var input2 string = "255.100.50.0"
	output2 := d.Defang(input2)
	fmt.Println("input:", input2, "output:", output2)
}

File: cs/algorithms/DefangedIpAddress.cs
namespace Studying.Leetcode.Algorithms
{
    /// <summary>
    /// Description: https://leetcode.com/problems/defanging-an-ip-address/
    /// </summary>
    public class DefangedIpAddress : Studying.Leetcode.ILeetcodeProblem
    {
        public void Execute()
        {
            System.Console.WriteLine("Defanging an IP-address\n".ToUpper()); 

            // Example 1:
            // Input: address = "1.1.1.1"
            // Output: "1[.]1[.]1[.]1"
            string input1 = "1.1.1.1";
            string output1 = Defang(input1);
            System.Console.WriteLine("input: " + input1); 
            System.Console.WriteLine("output: " + output1); 

            // Example 2:
            // Input: address = "255.100.50.0"
            // Output: "255[.]100[.]50[.]0"
            string input2 = "255.100.50.0";
            string output2 = Defang(input2);
            System.Console.WriteLine("input: " + input2); 
            System.Console.WriteLine("output: " + output2); 
        }

        private string Defang(string input)
        {
            // Constraints:
            // The given address is a valid IPv4 address.
            System.Net.IPAddress address;
            if (!System.Net.IPAddress.TryParse(input, out address)) 
                throw new System.Exception("Violated constraint: The given address is a valid IPv4 address"); 

            // Option 1: 
            // return input.Replace(".", "[.]"); 

            // Option 2: 
            string tmp = string.Empty; 
            for (int i = 0; i < input.Length; i++) 
                tmp += input[i] == '.' ? "[.]" : input[i].ToString(); 
            return tmp; 
        }
    }
}